monitoring/daily_reporter/src/elasticsearch.py

```diff
@@ -129,4 +131,48 @@ def get_interesting_ingests(es_client, *, index_name, days_to_fetch):
             # Nothing in hits, no results to find
             pass
 
+    # Look up the last 10,000 failures.  We'll discard all failures that
+    # are explainable; the idea is to highlight any failures that have
+    # an inexplicable failure reason, to keep them coming up until we fix
+    # them or explicitly discard them.
+    resp = es_client.request(
+        method="GET",
+        url=f"/{index_name}/_search",
+        json={
+            "query": {"bool": {"filter": [{"terms": {"status.id": ["failed"]}}]}},
+            "_source": [
+                "id",
+                "lastModifiedDate",
+                "space.id",
+                "status.id",
+                "bag.info.externalIdentifier",
+                "bag.info.version",
+                "createdDate",
+                "events.createdDate",
+                "events.description",
+            ],
+            # We can retrieve up to 10000 documents in one request.
+            "size": 10000,
+            "sort": [{"id": "desc"}],
+        },
+    )
+    for hit in resp.json().get("hits", {}).get("hits", []):
+        failed_ingest = _parse_ingest_from_hit(hit)
+
+        if is_known_failure(hit["_id"]):
+            continue
+
+        if get_dev_status(failed_ingest) in {
+            "failed (user error)",
+            "failed (known error)",
+        }:
+            continue
+        else:
+            ingests[hit["_id"]] = failed_ingest
+
+        # If there are more than 10000 results, that suggests there are more
+        # that we didn't get.  Flag this to the caller.
+        if resp.json()["hits"]["total"]["value"] >= 10000:
```

Comment:
You could use [scroll](https://elasticsearch-py.readthedocs.io/en/7.10.0/api.html?highlight=scroll#elasticsearch.Elasticsearch.scroll) if you wanted more

Reply:
\*nod*

That means adding functionality that I'd have to test and maintain, and right now it's not an issue – we never actually hit the 10000 limit, but I don't want it to fail silently if we do.